#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"

WorldInfo {
  basicTimeStep 16
}
Viewpoint {
  orientation -0.03352687850011758 -0.033286692643830475 -0.9988833488004861 3.4840320584243094
  position -2.85512893325975 -9.028744087223714 -0.0702578121493906
  follow "Little Bicycle V2"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 0 -0.1
  floorSize 10 20
  floorTileSize 10 20
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "textures/track001.jpeg"
      ]
    }
    roughness 1
    metalness 0
  }
}
Robot {
  translation -3.71003 -8.51 -0.0450894
  rotation 0 0 1 3.14158
  children [
    DEF frame Transform {
      scale 0.001 0.001 0.001
      children [
        CadShape {
          url [
            "obj/Little Bicycle V2 - Frame/frame.obj"
          ]
        }
      ]
    }
    DEF rear_wheel HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0 0.083 0
      }
      device [
        RotationalMotor {
          name "motor::wheel"
          maxVelocity 200
          multiplier 2
        }
      ]
      endPoint Solid {
        translation 0 0.083 0
        rotation 1 0 0 0
        children [
          DEF wheel Transform {
            scale 0.001 0.001 0.001
            children [
              CadShape {
                url [
                  "obj/Little Bicycle - Rear Wheel/r_wheel.obj"
                ]
              }
            ]
          }
        ]
        boundingObject Pose {
          rotation 0 1 0 1.5708
          children [
            Cylinder {
              height 0.015
              radius 0.055
              subdivision 72
            }
          ]
        }
        physics Physics {
          centerOfMass [
            0 0 0
          ]
        }
      }
    }
    DEF crank HingeJoint {
      jointParameters HingeJointParameters {
      }
      device [
        RotationalMotor {
          name "motor::crank"
          maxVelocity 100
        }
      ]
      endPoint Solid {
        rotation 1 0 0 0
        children [
          DEF crank Transform {
            scale 0.001 0.001 0.001
            children [
              CadShape {
                url [
                  "obj/Little Bicycle - Crank/crank.obj"
                ]
              }
            ]
          }
          DEF pedal HingeJoint {
            jointParameters HingeJointParameters {
              anchor -0.0375 -0.03 0
            }
            endPoint Solid {
              translation -0.0375 -0.03 0
              rotation 1 0 0 0
              children [
                DEF pedal Transform {
                  scale 0.001 0.001 0.001
                  children [
                    CadShape {
                      url [
                        "obj/Little Bicycle - Pedal/pedal.obj"
                      ]
                    }
                  ]
                }
              ]
              boundingObject Pose {
                children [
                  Shape {
                    geometry Box {
                      size 0.02 0.02 0.006
                    }
                  }
                ]
              }
              physics Physics {
              }
            }
          }
          DEF pedal HingeJoint {
            jointParameters HingeJointParameters {
              anchor 0.0375 0.03 0
            }
            endPoint Solid {
              translation 0.0375 0.03 0
              rotation 1 0 0 0
              children [
                DEF pedal Transform {
                  scale 0.001 0.001 0.001
                  children [
                    CadShape {
                      url [
                        "obj/Little Bicycle - Pedal/pedal.obj"
                      ]
                    }
                  ]
                }
              ]
              name "solid(1)"
              boundingObject Pose {
                children [
                  Shape {
                    geometry Box {
                      size 0.02 0.02 0.006
                    }
                  }
                ]
              }
              physics Physics {
              }
            }
          }
        ]
        name "solid(2)"
        boundingObject Group {
          children [
            Pose {
              translation -0.0238 -0.015 0
              children [
                Shape {
                  geometry Box {
                    size 0.005 0.04 0.01
                  }
                }
              ]
            }
            Pose {
              translation 0.0238 0.015 0
              children [
                Shape {
                  geometry Box {
                    size 0.005 0.04 0.01
                  }
                }
              ]
            }
          ]
        }
        physics Physics {
        }
      }
    }
    DEF Handlebars_and_Fork HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0 -0.0818 0.102
      }
      device [
        RotationalMotor {
          name "handlebars motor"
        }
      ]
      endPoint Solid {
        translation 0 -0.0818 0.102
        children [
          DEF fork Transform {
            scale 0.001 0.001 0.001
            children [
              CadShape {
                url [
                  "obj/Little Bicycle V2 - Handlebars and Fork/fork.obj"
                ]
              }
            ]
          }
          DEF wheel HingeJoint {
            jointParameters HingeJointParameters {
              anchor 0 -0.037 -0.102
            }
            endPoint Solid {
              translation 0 -0.037 -0.102
              rotation 1 0 0 0
              children [
                DEF wheel Transform {
                  scale 0.001 0.001 0.001
                  children [
                    CadShape {
                      url [
                        "obj/Little Bicycle - Front Wheel/f_wheel.obj"
                      ]
                    }
                  ]
                }
              ]
              boundingObject Pose {
                rotation 0 1 0 1.5708
                children [
                  Cylinder {
                    height 0.015
                    radius 0.055
                    subdivision 72
                  }
                ]
              }
              physics Physics {
              }
            }
          }
        ]
        name "solid(1)"
        boundingObject Group {
          children [
            Pose {
              translation 0.0125 -0.024 -0.065
              rotation -1 0 0 0.34909999999999985
              children [
                Shape {
                  geometry Box {
                    size 0.005 0.005 0.093
                  }
                }
              ]
            }
            Pose {
              translation -0.0125 -0.024 -0.065
              rotation 1 0 0 -0.3491
              children [
                Shape {
                  geometry Box {
                    size 0.005 0.005 0.093
                  }
                }
              ]
            }
            Pose {
              translation 0.0125 -0.037 -0.102
              rotation 0 1 0 1.5708
              children [
                Shape {
                  geometry Cylinder {
                    height 0.005
                    radius 0.01
                  }
                }
              ]
            }
            Pose {
              translation -0.0125 -0.037 -0.102
              rotation 0 1 0 1.5708
              children [
                Shape {
                  geometry Cylinder {
                    height 0.005
                    radius 0.01
                  }
                }
              ]
            }
            Pose {
              translation 0 -0.008 -0.022
              rotation 1 0 0 -0.3491
              children [
                Shape {
                  geometry Cylinder {
                    height 0.01
                    radius 0.0125
                  }
                }
              ]
            }
            Pose {
              translation 0 0.0035 0.0085
              rotation 1 0 0 -0.3491
              children [
                Shape {
                  geometry Cylinder {
                    height 0.075
                    radius 0.0025
                  }
                }
              ]
            }
            Pose {
              translation -0.062 0.0121 0.05445
              rotation 0.3796282240312281 0.893242880073478 0.24083099617145198 1.5708
              children [
                Shape {
                  geometry Cylinder {
                    height 0.04
                    radius 0.006
                  }
                }
              ]
            }
            Pose {
              translation 0.062 0.0121 0.05445
              rotation -0.06982912769991387 -0.9687746404465684 0.23790668118464223 1.5708
              children [
                Shape {
                  geometry Cylinder {
                    height 0.04
                    radius 0.006
                  }
                }
              ]
            }
            Pose {
              translation 0 0.0045 0.0441
              rotation 1 0 0 -0.3491
              children [
                Shape {
                  geometry Box {
                    size 0.03 0.005 0.005
                  }
                }
              ]
            }
            Pose {
              translation -0.0203 0.0066 0.0498
              rotation -0.3866558708915664 0.9082106967389628 -0.16015794652158893 0.855791
              children [
                Shape {
                  geometry Box {
                    size 0.0201 0.005 0.005
                  }
                }
              ]
            }
            Pose {
              translation 0.0203 0.0066 0.0498
              rotation 0.3866558708915664 0.9082106967389628 -0.16015794652158893 -0.8557953071795863
              children [
                Shape {
                  geometry Box {
                    size 0.0201 0.005 0.005
                  }
                }
              ]
            }
            Pose {
              translation 0.036 0.0087 0.0556
              rotation 1 0 0 -0.3491
              children [
                Shape {
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
            Pose {
              translation -0.036 0.0087 0.0556
              rotation 1 0 0 -0.3491
              children [
                Shape {
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
          ]
        }
        physics Physics {
        }
      }
    }
    Camera {
      translation -7.27661e-08 -0.086891 0.200299
      rotation -0.25056306820553154 -0.25056206820525934 0.9351132545462786 -1.6378353071795866
      fieldOfView 2
      width 480
      height 320
      antiAliasing TRUE
    }
    Display {
      width 480
      height 320
    }
    InertialUnit {
      translation 0 0 0.05
      rotation 1 0 0 0
      name "imu"
    }
  ]
  name "Little Bicycle V2"
  boundingObject Group {
    children [
      Pose {
        translation 0 -0.026 0.0916
        rotation 1 0 0 -0.3491
        children [
          Shape {
            geometry Box {
              size 0.01 0.11 0.01
            }
          }
        ]
      }
      Pose {
        translation 0 0.0162 0.056
        rotation 1 0 0 -0.2793
        children [
          Shape {
            geometry Box {
              size 0.01 0.01 0.135
            }
          }
        ]
      }
      Pose {
        translation 0 -0.045 0.056
        rotation 1 0 0 0.6632
        children [
          Shape {
            geometry Box {
              size 0.01 0.01 0.135
            }
          }
        ]
      }
      Pose {
        translation 0.0125 0.0529 0.038
        rotation 1 0 0 0.6981
        children [
          Shape {
            geometry Box {
              size 0.005 0.005 0.1
            }
          }
        ]
      }
      Pose {
        translation -0.0125 0.0529 0.038
        rotation 1 0 0 0.6981
        children [
          Shape {
            geometry Box {
              size 0.005 0.005 0.11
            }
          }
        ]
      }
      Pose {
        translation -0.0125 0.046 -2.1684e-19
        children [
          Shape {
            geometry Box {
              size 0.005 0.08 0.005
            }
          }
        ]
      }
      Pose {
        translation 0.0125 0.046 -2.1684e-19
        children [
          Shape {
            geometry Box {
              size 0.005 0.08 0.005
            }
          }
        ]
      }
      Pose {
        translation 0 0.045 0.1221
        children [
          Shape {
            geometry Cylinder {
              height 0.01
              radius 0.02405
            }
          }
        ]
      }
      Pose {
        translation 0 0.018 0.1221
        children [
          Shape {
            geometry Cylinder {
              height 0.01
              radius 0.015
            }
          }
        ]
      }
      Pose {
        rotation 0 1 0 1.5708
        children [
          Shape {
            geometry Cylinder {
              height 0.04
              radius 0.01
            }
          }
        ]
      }
      Pose {
        translation 0 -0.082 0.102
        rotation 1 0 0 -0.3491
        children [
          Shape {
            geometry Cylinder {
              height 0.035
              radius 0.01
            }
          }
        ]
      }
      Pose {
        translation -0.0125 0.083 0
        rotation 0 1 0 1.5708
        children [
          Shape {
            geometry Cylinder {
              height 0.005
              radius 0.01
            }
          }
        ]
      }
      Pose {
        translation 0.0125 0.083 0
        rotation 0 1 0 1.5708
        children [
          Shape {
            geometry Cylinder {
              height 0.005
              radius 0.01
            }
          }
        ]
      }
      Pose {
        translation 0 0.0212 0.0747
        rotation 1 0 0 -0.3491
        children [
          Shape {
            geometry Cylinder {
              height 0.01
              radius 0.0125
            }
          }
        ]
      }
    ]
  }
  physics Physics {
  }
  controller "little_bicycle_P_V2"
  supervisor TRUE
}
